### ISSUE TITLE
False-positive while_immutable_condition lint when condition is static mut

### ISSUE THREAD

**Issue by SamokhinIlya** [rust-lang-nursery/rust-clippy#2710]
The following code triggers while_immutable_condition lint, however, RUNNING is mutated:
```Rust
static mut RUNNING: bool = true;

fn main() {
    unsafe {
        while RUNNING {
            RUNNING = false;
        }
    }
}
```

```
error: Variable in the condition are not mutated in the loop body. This either leads to an infinite or to a never running loop.
 --> src\main.rs:5:15
  |
5 |         while RUNNING {
  |               ^^^^^^^
  |
  = note: #[deny(while_immutable_condition)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.195/index.html#while_immutable_condition
```
Not sure if this is related to [#2584](https://github.com/rust-lang-nursery/rust-clippy/issues/2584) or [#2584](https://github.com/rust-lang-nursery/rust-clippy/issues/2596).
